version: "3.3"

services:
  traefik:
    image: traefik:v2.1.3
    ports:
      - 80:80
      - 8080:8080
      - 443:443
    command:
      - --api.insecure=true
      - --api.dashboard=true
      - --api.debug=true
      - --log.level=DEBUG
      - --accesslog=true
      - --providers.docker=true
      - --providers.docker.swarmMode=true
      - --providers.docker.exposedbydefault=false
      - --providers.docker.network=proxy
      - --entrypoints.web.address=:80
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - proxy
    deploy:
      placement:
        constraints: [node.role == manager]
      restart_policy:
        condition: on-failure
      labels:
        - traefik.enable=true
        - traefik.http.routers.api.rule=Host(`${DOMAIN}`)
        - traefik.http.routers.api.service=api@internal
      replicas: 1

networks:
  proxy:
    external: true
